example/targets/neal_funnel.jl

```diff
@@ -0,0 +1,61 @@
+using Distributions, Random
+
+"""
+    Funnel{T<:Real}
+
+Multidimensional Neal's Funnel distribution
+
+# Fields 
+- 'dim::Int': Dimension of the distribution, must be >= 2
+- 'μ::T': Mean of the first dimension
+- 'σ::T': Standard deviation of the first dimension, must be > 0
+
+# Explanation
+
+The Neal's Funnel distribution is a p-dimensional distribution with a funnel shape, 
+originally proposed by Radford Neal in [2]. 
+The marginal distribution of "x₁" is Gaussian with mean "μ" and standard
+deviation "σ". The conditional distribution of "x₂, …, xₚ" given "x₁" is independent 
+Gaussian distribution with mean 0 and standard deviation "exp(x₁/2)". 
+The generative process is given by
+"x₁ ∼ N(μ, σ²), x₂, ..., xₚ ∼ N(0, exp(x₁))"
+
+
+# Reference
+[1] Stan User’s Guide: 
+https://mc-stan.org/docs/2_18/stan-users-guide/reparameterization-section.html#ref-Neal:2003
+[2] Radford Neal 2003. “Slice Sampling.” Annals of Statistics 31 (3): 705–67.
+"""
+struct Funnel{T<:Real} <: ContinuousMultivariateDistribution
+    dim::Int      # Dimension
+    μ::T          # Mean of the first dimension
+    σ::T          # sd of the first dimension
+    function Funnel{T}(dim::Int, μ::T, σ::T) where {T<:Real}
+        dim >= 2 || error("dim must be >= 2")
+        σ > 0 || error("σ must be > 0")
+        return new{T}(dim, μ, σ)
+    end
+end
+Funnel(dim::Int, μ::T, σ::T) where {T<:Real} = Funnel{T}(dim, μ, σ)
+Funnel(dim::Int, σ::T) where {T<:Real} = Funnel{T}(dim, zero(T), σ)
+Funnel(dim::Int) = Funnel(dim, 0.0, 9.0)
+
+Base.length(p::Funnel) = p.dim
+Base.eltype(p::Funnel{T}) where {T<:Real} = T
+Distributions.sampler(p::Funnel) = p
```

Comment:
For funnel I think we do need as per: https://juliastats.org/Distributions.jl/v0.14/extends.html#Create-a-Multivariate-Distribution-1

Reply:
But it has a default implementation that is sufficient, see Distributions.jl docs on `sampler` here: https://juliastats.org/Distributions.jl/v0.14/extends.html#Distributions.sampler-Tuple{Distributions.Distribution}

In particular:

> If such a sampler exists, it can be provide with this sampler method, which would be used for batch sampling. The general fallback is sampler(d::Distribution) = d.

So it's only necessary to implement in the scenario where this default implementation isn't sufficient:)

Reply:
```suggestion
```